### PR TITLE
chore: Bump otel modules to v1.30.0/v0.55.0

### PR DESCRIPTION
**Pull request by codingllama**
Monthly dependency update batch, otel version.

* https://github.com/open-telemetry/opentelemetry-go/releases/tag/v1.30.0

### CODE REVIEW

**Review by tigrato — APPROVED** [review]

**Review by marcoandredinis — APPROVED** [review]

**Comment by codingllama** [comment]
See #46973

- #46973

**Comment by codingllama** [comment]
Thanks for the quick reviews!

File: api/go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/gravitational/teleport/api
 
-go 1.21
+go 1.22
```
Comment (hugoShaka) [inline api/go.mod:3]: Is this intentional?

Comment (codingllama) [reply api/go.mod:3]: Otel now requires Go 1.22, so this gets updated automatically. Unavoidable if we want to move certain dependencies forward.

Comment (codingllama) [reply api/go.mod:3]: (Well spotted, btw. I meant to add a comment about it but it escaped me.(

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**